src/evict.c

```diff
@@ -382,6 +382,10 @@ int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *lev
     /* We may return ASAP if there is no need to compute the level. */
     int return_ok_asap = !server.maxmemory || mem_reported <= server.maxmemory;
     if (return_ok_asap && !level) return C_OK;
+    if (!server.maxmemory && level) {
+        *level = 0;
+        return C_OK;
+    }
```

Comment:
The `return_ok_asap` variable doesn't make things clear anymore. Maybe it's simpler to get rid of it?

```suggestion
    if (!server.maxmemory) {
        if (level) *level = 0;
        return C_OK;
    }
    if (mem_reported <= server.maxmemory && !level) return C_OK;
```

... and below, change `if (return_ok_asap) return C_OK;` to `if (mem_reported <= server.maxmemory) return C_OK;`